### PR TITLE
✨ Improve kuadio ad network optional params data-ref

### PR DESCRIPTION
**Pull request by camel2243**
# Instructions:

- Pick a meaningful title for your pull request. (Use sentence case.)
  - Prefix the title with an emoji to identify what is being done. (Copy-paste the emoji itself (not the :code:) from the list below.)
  - Do not overuse punctuation in the title (like `(chore):`).
  - If it is helpful, use a simple prefix (like `ProjectX: Implement some feature`).
- Enter a succinct description that says why the PR is necessary, and what it does.
  - Mention the GitHub issue that is being addressed by the pull request.
  - The keywords `Fixes`, `Closes`, or `Resolves` followed the issue number will automatically close the issue.

# Example of a good description:

- Implement aspect X
- Leave out feature Y because of A
- Improve performance by B
- Improve accessibility by C

# Emojis for categorizing pull requests:

✨ New feature (`:sparkles:`)  
🐛 Bug fix (`:bug:`)  
🔥 P0 fix (`:fire:`)  
✅ Tests (`:white_check_mark:`)  
🚀 Performance improvements (`:rocket:`)  
🖍 CSS / Styling (`:crayon:`)  
♿ Accessibility (`:wheelchair:`)  
🌐 Internationalization (`:globe_with_meridians:`)  
📖 Documentation (`:book:`)  
🏗 Infrastructure / Tooling / Builds / CI (`:building_construction:`)  
⏪ Reverting a previous change (`:rewind:`)  
♻️ Refactoring (like moving around code w/o any changes) (`:recycle:`)  
🚮 Deleting code (`:put_litter_in_its_place:`)

### CODE REVIEW

**Comment by camel2243** [comment]
We add optional params `data-ref` however percy test is failed, it's should not affect the test.

Could you check status and review our commits?

Best regards.

**Comment by camel2243** [comment]
@lannka  Could you review this pr?

File: ads/kuadio.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ export function kuadio(global, data) {
   const e = global.document.createElement('div');
   e.className = '_pvmax_recommend';
   e.setAttribute('data-widget-id', data.widgetId);
-  e.setAttribute('data-ref', window.context.canonicalUrl);
+  e.setAttribute('data-ref', data.ref || window.context.canonicalUrl);
```
Comment (lannka) [inline ads/kuadio.js]: use `global.context.canonicalUrl`

**Comment by camel2243** [comment]
@lannka I fixed the code. Could you review it again?

**Comment by lannka** [comment]
@camel2243 it looks all good. will merge it.

@rsimha I don't think this PR would change bundle size , as it only touches 3P code. Can you help to check why it's failing the bundle size check?

**Comment by rsimha** [comment]
@danielrozenberg is fixing the bundle-size baseline logic. Feel free to ignore the check for now.

**Comment by lannka** [comment]
@camel2243 there is a lint issue. please fix the travis presubmit check.

**Comment by lannka** [comment]
awesome! thanks. merging

**Pull request merged**